envoy/auth_server/main.go

```diff
@@ -140,8 +85,60 @@ func main() {
 	// Register envoy proto server
 	envoy_auth.RegisterAuthorizationServer(grpcServer, authHandler)
 
-	fmt.Printf("Auth server starting on port %d...\n", port)
+	fmt.Printf("Auth server starting on port %d...\n", config.Port)
 	if err = grpcServer.Serve(listen); err != nil {
 		panic(err)
 	}
 }
+
+/* -------------------- Gateway Init Helpers -------------------- */
+
+// getConfigPath returns the full path to the config file
+// based on the current working directory of the executable.
+//
+// For example if the binary is in:
+// - `/app` the full config path will be `/app/config/.config.yaml`
+// - `./bin` the full config path will be `./bin/config/.config.yaml`
+func getConfigPath() (string, error) {
+	var configPath string
+
+	// The config path can be overridden using the `-config` flag.
+	flag.StringVar(&configPath, "config", "", "override the default config path")
+	flag.Parse()
+	if configPath != "" {
+		return configPath, nil
+	}
+
+	// Otherwise, use the default config path based on the executable path
+	exeDir, err := os.Executable()
+	if err != nil {
+		return "", fmt.Errorf("failed to get executable path: %v", err)
+	}
+
+	configPath = filepath.Join(filepath.Dir(exeDir), defaultConfigPath)
+
+	return configPath, nil
+}
+
+// connectGRPC connects to the gRPC server for the GatewayEndpoints service
+// and returns a gRPC client connection.
+func connectGRPC(hostPort string, useInsecureCredentials bool) (*grpc.ClientConn, error) {
+	var transport grpc.DialOption
```

Comment:
Consider changing this to `credentials` and doing `transport = grpc.WithTransportCredentials(credentials)` on th second last line.